lib/async_hooks.js

```diff
@@ -353,6 +353,9 @@ class AsyncLocalStorage {
   }
 }
 
+// Create deprecated enterWith alias
+AsyncLocalStorage.prototype.enterWith = AsyncLocalStorage.prototype.set;
```

Comment:
Can you emit warning when `enterWith` is called?

Reply:
Util.deprecate should do the trick

Reply:
Is there a reason for not using a doc only deprecation here - at least for a while? Deprecation logs may break apps using the console output.

It's just a about an inoptimum naming not a functional problem which results in some risk/problem.

Reply:
I'd argue we want the deprecation logs to show for a few versions before we remove `enterWith `. Alternativelly, we can keep `enterWith` and doc it as an alias for `set`, but if we want it out, we need to make sure we never break code with `. enterWith is not a function`

Reply:
I'm fine with keeping it as documented alias like I did it here. My main intention is that looking into real world use cases as in the above linked PR the name looks better to me and results in easier to read code.

But `enterWith` is not that bad that I would force people to change. can't tell how many users there are actually.

Reply:
Maybe an advantage of `enterWith` is that it's "unique" and we can easily search for its usage in GitHub and other open source databases.

Reply:
As mentioned in the main thread, I changed my mind, I think we should keep `enterWith` as a forever alias due to its use in existing APMs and we don't want to break any of their customers on whatever version of node x version of the APM.

Reply:
Yeah, this is used extensively at Datadog currently so I definitely would not want to go straight to warning deprecation. We should have a doc deprecation phase, an eventual warning phase, and then _maybe_ removal.

As I stated further up in the issue though, I'm working on making this API (hopefully) obsolete so not sure changes the this will really matter much.